### DISCUSSION TITLE
Some questions on binary version with "--noconsole" option

### DISCUSSION THREAD

**Discussion by hwy1992129** [pyinstaller/pyinstaller#7881] [PyInstaller] [answered]
Hello,

I've been using PyInstaller to convert my Python scripts into binaries, particularly with the --noconsole option. While the conversion process is understandably lengthy due to dependencies, my primary concern lies in debugging the resulting binary.

When I use the --noconsole option, I'm unable to see errors, which makes it challenging to diagnose issues. Interestingly, certain errors arise only when the console is disabled. When I use the console option, those errors don't manifest. I think I can create a log in binary version. However, is there a recommended way to effectively debug these issues in the binary version without the console?

Moreover, each conversion is time-consumed..... Once I check the log and fix the bugs in script, I have to convert it to binary again to test it.....

**Comment by rokm** [comment, marked as answer]
If the error in `noconsole` executable (assuming we are talking about Windows here) results in an unhandled exception, PyInstaller's bootloader should pop up an error window with information about exception and its traceback. 

If the error is handled but for example prints a warning to stdout/stderr (for example via `logging`) then you won't see it due to lack of console. So if you want to capture warnings that would otherwise go to stderr, you will need to redirect that to a log file.

> Interestingly, certain errors arise only when the console is disabled. When I use the console option, those errors don't manifest. 

Then your code or the 3rd party code you are using is being naive and expects that  `sys.stderr` and `sys.stdout` are available, whereas they are not and are `None` in noconsole mode. For example, trying to call `sys.stderr.flush()` without first checking that `sys.stderr` is not `None` will result in an attribute error.
